### ISSUE TITLE
Issues Autantification

### ISSUE THREAD

**Issue by Geoff-Hz** [boredazfcuk/docker-icloudpd#105]
Hi,
I have an issues when a insert my password.
I am working on Docker from my Synology.

here is the code:
`root@TheBoss:~# docker create \
> --name icloudpd \
> --network bridge \
> --restart unless-stopped \
> -e user=\<sensitive info removed\> \
> -e user_id=1026 \
> -e group=users \
> -e group_id=100 \
> -e apple_id=\<sensitive info removed\> \
> -e authentication_type=web \
> -e folder_structure={:%Y} \
> -e auto_delete=true \
> -e synchronisation_interval=86400 \
> -e TZ=Europe/Paris \
> -v /volume1/docker/iCloud/config:/config \
> -v /volume1/photo/iCloud:/home/\<sensitive info removed\>/iCloud \
> boredazfcuk/icloudpd
83b48a6b2b3d22f1bb562a014b82dc3e47876a64c362a53b72335b359b9f549e
root@TheBoss:~# docker exec -it icloudpd sync-icloud.sh --Initialise
Error response from daemon: Container 83b48a6b2b3d22f1bb562a014b82dc3e47876a64c3                                                                 62a53b72335b359b9f549e is not running
root@TheBoss:~# docker start icloudpd
icloudpd
root@TheBoss:~# docker exec -it icloudpd sync-icloud.sh --Initialise

2021-09-21 11:33:12 INFO     ***** boredazfcuk/icloudpd container for icloud_pho                                                                 to_downloader started *****
2021-09-21 11:33:12 INFO     ***** /usr/local/bin/sync-icloud.sh date: 2021/09/1                                                                 6_00:40 *****
2021-09-21 11:33:12 INFO     ***** /usr/local/bin/sync-icloud.sh hash: 5a180ee43                                                                 ca07a00a70080c9d969dcad *****
2021-09-21 11:33:12 INFO     Alpine Linux 3.13.5
2021-09-21 11:33:12 INFO     Python version: 3.8.10
2021-09-21 11:33:12 INFO     icloudpd version: 1.7.2
2021-09-21 11:33:26 INFO     pyicloud-ipd version: 0.10.1
2021-09-21 11:33:34 INFO     Running user id: 0
2021-09-21 11:33:34 INFO     Running group id: 0
2021-09-21 11:33:34 INFO     Local user: <sensitive info removed>:1026
2021-09-21 11:33:34 INFO     Local group: users:100
2021-09-21 11:33:34 INFO     Force GID: False
2021-09-21 11:33:34 INFO     LAN IP Address: 172.17.0.3
2021-09-21 11:33:34 INFO     Apple ID: <sensitive info removed>
2021-09-21 11:33:34 INFO     Authentication Type: web
2021-09-21 11:33:34 INFO     Cookie path: /config/<sensitive info removed>
2021-09-21 11:33:34 INFO     Cookie expiry notification period: 7
2021-09-21 11:33:34 INFO     Download destination directory: /home/<sensitive info removed>/iCloud
2021-09-21 11:33:34 INFO     Folder structure: {:%Y}
2021-09-21 11:33:34 INFO     Directory permissions: 750
2021-09-21 11:33:34 INFO     File permissions: 640
2021-09-21 11:33:34 INFO     Synchronisation interval: 86400
2021-09-21 11:33:34 INFO     Synchronisation delay (minutes): 0
2021-09-21 11:33:34 INFO     Time zone: Europe/Paris
2021-09-21 11:33:34 INFO     Set EXIF date/time: False
2021-09-21 11:33:34 INFO     Auto delete: true
2021-09-21 11:33:34 INFO     Photo size: original
2021-09-21 11:33:34 INFO     Skip download check: False
2021-09-21 11:33:34 INFO     Skip live photos: False
2021-09-21 11:33:34 INFO     Number of most recently added photos to download: Download All Photos
2021-09-21 11:33:34 INFO     Stop downloading when prexisiting files count is: Download All Photos
2021-09-21 11:33:34 INFO     Live photo size: original
2021-09-21 11:33:34 INFO     Skip videos: False
2021-09-21 11:33:34 INFO     Convert HEIC to JPEG: False
2021-09-21 11:33:34 INFO     JPEG conversion quality: 90
2021-09-21 11:33:34 INFO     Creating directory: /home/<sensitive info removed>/.local/share/
2021-09-21 11:33:34 INFO     Creating symbolic link: /home/<sensitive info removed>/.local/share/python_keyring/ to: /config/python_keyring/ directory
2021-09-21 11:33:34 INFO     Script launch parameters: --Initialise
2021-09-21 11:33:34 INFO     Group, users:100, already created
2021-09-21 11:33:34 INFO     Creating user <sensitive info removed>:1026
2021-09-21 11:33:34 INFO     Set owner, <sensitive info removed>, on iCloud directory, if required
2021-09-21 11:33:34 INFO     Set group, users, on iCloud directory, if required
2021-09-21 11:33:34 INFO     Correct owner on icloudpd temp directory, if required
2021-09-21 11:33:34 INFO     Correct group on icloudpd temp directory, if required
2021-09-21 11:33:34 INFO     Correct owner on config directory, if required
2021-09-21 11:33:34 INFO     Correct group on config directory, if required
2021-09-21 11:33:34 INFO     Correct owner on keyring directory, if required
2021-09-21 11:33:34 INFO     Correct group on keyring directory, if required
2021-09-21 11:33:34 INFO     Set 750 permissions on iCloud directories, if required
2021-09-21 11:33:34 INFO     Set 640 permissions on iCloud files, if required
2021-09-21 11:33:34 INFO     Configure password
2021-09-21 11:33:34 INFO     Adding password to keyring file: /config/python_keyring/keyring_pass.cfg
Enter iCloud password for <sensitive info removed>:
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 353, in connect
    conn = self._new_conn()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 181, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7fa4cbdd3df0>: Failed to establish a new connection: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/lib/python3.8/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='setup.icloud.com', port=443): Max retries exceeded with url: /setup/ws/1/login?clientBuildNumber=17DHotfix5&clientMasteringNumber=17DHotfix5&ckjsBuildVersion=17DProjectDev77&ckjsVersion=2.0.5&clientId=12092C00-1ABF-11EC-A998-0242AC110003 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fa4cbdd3df0>: Failed to establish a new connection: [Errno -3] Try again'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/icloud", line 33, in <module>
    sys.exit(load_entry_point('pyicloud-ipd==0.10.1', 'console_scripts', 'icloud')())
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/cmdline.py", line 200, in main
    api = pyicloud_ipd.PyiCloudService(
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 193, in __init__
    self.authenticate()
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 209, in authenticate
    req = self.session.post(
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 590, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 68, in request
    response = super(PyiCloudSession, self).request(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='setup.icloud.com', port=443): Max retries exceeded with url: /setup/ws/1/login?clientBuildNumber=17DHotfix5&clientMasteringNumber=17DHotfix5&ckjsBuildVersion=17DProjectDev77&ckjsVersion=2.0.5&clientId=12092C00-1ABF-11EC-A998-0242AC110003 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fa4cbdd3df0>: Failed to establish a new connection: [Errno -3] Try again'))
2021-09-21 11:34:13 INFO     Correct owner on config directory, if required
2021-09-21 11:34:13 INFO     Correct group on config directory, if required
2021-09-21 11:34:13 INFO     Generate 2FA cookie using password stored in keyring file.
iCloud Password:
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/authentication.py", line 29, in authenticate
    icloud = pyicloud_ipd.PyiCloudService(
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 141, in __init__
    password = get_password_from_keyring(apple_id)
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/utils.py", line 40, in get_password_from_keyring
    raise NoStoredPasswordAvailable(
pyicloud_ipd.exceptions.NoStoredPasswordAvailable: No pyicloud password for geoffroy.mail@gmail.com could be found in the system keychain.  Use the `--store-in-keyring` command-line option for storing a password for this username.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 353, in connect
    conn = self._new_conn()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 181, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f86de0ee2b0>: Failed to establish a new connection: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/lib/python3.8/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='setup.icloud.com', port=443): Max retries exceeded with url: /setup/ws/1/login?clientBuildNumber=17DHotfix5&clientMasteringNumber=17DHotfix5&ckjsBuildVersion=17DProjectDev77&ckjsVersion=2.0.5&clientId=43DB231E-1ABF-11EC-81B1-0242AC110003 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f86de0ee2b0>: Failed to establish a new connection: [Errno -3] Try again'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/icloudpd", line 33, in <module>
    sys.exit(load_entry_point('icloudpd==1.7.2', 'console_scripts', 'icloudpd')())
  File "/usr/lib/python3.8/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/base.py", line 255, in main
    icloud = authenticate(
  File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/authentication.py", line 36, in authenticate
    icloud = pyicloud_ipd.PyiCloudService(
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 193, in __init__
    self.authenticate()
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 209, in authenticate
    req = self.session.post(
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 590, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 68, in request
    response = super(PyiCloudSession, self).request(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='setup.icloud.com', port=443): Max retries exceeded with url: /setup/ws/1/login?clientBuildNumber=17DHotfix5&clientMasteringNumber=17DHotfix5&ckjsBuildVersion=17DProjectDev77&ckjsVersion=2.0.5&clientId=43DB231E-1ABF-11EC-81B1-0242AC110003 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f86de0ee2b0>: Failed to establish a new connection: [Errno -3] Try again'))
2021-09-21 11:35:37 INFO     Web cookie generated. Sync should now be successful.
`

Can you help me?
Thanks

**Comment by Geoff-Hz** [comment]
Also, the file /config/python_keyring/keyring_pass.cfg doesn't exist (ok for the directory python_keying)

**Comment by boredazfcuk** [comment]
Somebody else has this exact same issue, but I was struggling to pin it down. I'm glad there's two people having it now!!

Can you update to the latest version and post another log. I've amended the logging info so it includes extra info to help me work out your issue.

**Comment by Geoff-Hz** [comment]
Hi, Thanks for your help.
here the new log:

root@TheBoss:~# docker create \
> --name icloudpd \
> --network bridge \
> --restart unless-stopped \
> -e user=Geoff \
> -e user_id=1026 \
> -e group=users \
> -e group_id=100 \
> -e apple_id=geoffroy.mail@gmail.com \
> -e authentication_type=web \
> -e folder_structure={:%Y} \
> -e auto_delete=true \
> -e synchronisation_interval=86400 \
> -e TZ=Europe/Paris \
> -v /volume1/docker/iCloud/config:/config \
> -v /volume1/photo/iCloud:/home/Geoff/iCloud \
> boredazfcuk/icloudpd
Unable to find image 'boredazfcuk/icloudpd:latest' locally
latest: Pulling from boredazfcuk/icloudpd
540db60ca938: Pull complete
6c2af49a26ad: Pull complete
45667911aeaf: Pull complete
657af178745a: Pull complete
Digest: sha256:45293fd8146d8d40080b75d416aaece9630f31ba2aa92e9b0ebf444feb07e0ee
Status: Downloaded newer image for boredazfcuk/icloudpd:latest
d204e635348e6a336871f99664994ded9ac6f55612801ecc7f817ee030fc8976
root@TheBoss:~# docker start icloudpd
icloudpd
root@TheBoss:~# docker exec -it icloudpd sync-icloud.sh --Initialise


2021-09-21 14:05:11 INFO     ***** boredazfcuk/icloudpd container for icloud_photo_downloader started *****
2021-09-21 14:05:11 INFO     ***** /usr/local/bin/sync-icloud.sh date: 2021/09/21_13:45 *****
2021-09-21 14:05:11 INFO     ***** /usr/local/bin/sync-icloud.sh hash: 4f7994ad43454bba6aabe4fc46788e50 *****
2021-09-21 14:05:12 INFO     Alpine Linux 3.13.5
2021-09-21 14:05:12 INFO     Python version: 3.8.10
2021-09-21 14:05:12 INFO     icloudpd version: 1.7.2
2021-09-21 14:05:19 INFO     pyicloud-ipd version: 0.10.1
2021-09-21 14:05:26 INFO     Running user id: 0
2021-09-21 14:05:26 INFO     Running group id: 0
2021-09-21 14:05:26 INFO     Local user: Geoff:1026
2021-09-21 14:05:26 INFO     Local group: users:100
2021-09-21 14:05:26 INFO     Force GID: False
2021-09-21 14:05:26 INFO     LAN IP Address: 172.17.0.3
2021-09-21 14:05:26 INFO     Default gateway: 172.17.0.1
2021-09-21 14:05:26 INFO     DNS server: 10.0.1.1
2021-09-21 14:05:26 ERROR    Cannot find icloud.com IP address. Please check your DNS settings
2021-09-21 14:05:31 ERROR    No route to icloud.com found. Please check your container's network settings
2021-09-21 14:05:31 INFO     Apple ID: geoffroy.mail@gmail.com
2021-09-21 14:05:31 INFO     Authentication Type: web
2021-09-21 14:05:31 INFO     Cookie path: /config/geoffroymailgmailcom
2021-09-21 14:05:31 INFO     Cookie expiry notification period: 7
2021-09-21 14:05:31 INFO     Download destination directory: /home/Geoff/iCloud
2021-09-21 14:05:31 INFO     Folder structure: {:%Y}
2021-09-21 14:05:31 INFO     Directory permissions: 750
2021-09-21 14:05:31 INFO     File permissions: 640
2021-09-21 14:05:31 INFO     Synchronisation interval: 86400
2021-09-21 14:05:31 INFO     Synchronisation delay (minutes): 0
2021-09-21 14:05:31 INFO     Time zone: Europe/Paris
2021-09-21 14:05:31 INFO     Set EXIF date/time: False
2021-09-21 14:05:31 INFO     Auto delete: true
2021-09-21 14:05:31 INFO     Photo size: original
2021-09-21 14:05:31 INFO     Skip download check: False
2021-09-21 14:05:31 INFO     Skip live photos: False
2021-09-21 14:05:31 INFO     Number of most recently added photos to download: Download All Photos
2021-09-21 14:05:31 INFO     Stop downloading when prexisiting files count is: Download All Photos
2021-09-21 14:05:31 INFO     Live photo size: original
2021-09-21 14:05:31 INFO     Skip videos: False
2021-09-21 14:05:31 INFO     Convert HEIC to JPEG: False
2021-09-21 14:05:31 INFO     JPEG conversion quality: 90
2021-09-21 14:05:31 INFO     Script launch parameters: --Initialise
2021-09-21 14:05:31 INFO     Group, users:100, already created
2021-09-21 14:05:31 INFO     User, Geoff:1026, already created
2021-09-21 14:05:31 INFO     Set owner, Geoff, on iCloud directory, if required
2021-09-21 14:05:31 INFO     Set group, users, on iCloud directory, if required
2021-09-21 14:05:31 INFO     Correct owner on icloudpd temp directory, if required
2021-09-21 14:05:31 INFO     Correct group on icloudpd temp directory, if required
2021-09-21 14:05:31 INFO     Correct owner on config directory, if required
2021-09-21 14:05:31 INFO     Correct group on config directory, if required
2021-09-21 14:05:31 INFO     Correct owner on keyring directory, if required
2021-09-21 14:05:31 INFO     Correct group on keyring directory, if required
2021-09-21 14:05:31 INFO     Set 750 permissions on iCloud directories, if required
2021-09-21 14:05:31 INFO     Set 640 permissions on iCloud files, if required
2021-09-21 14:05:31 INFO     Configure password
2021-09-21 14:05:31 INFO     Adding password to keyring file: /config/python_keyring/keyring_pass.cfg
Enter iCloud password for geoffroy.mail@gmail.com:
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 353, in connect
    conn = self._new_conn()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 181, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f43e794bb50>: Failed to establish a new connection: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/lib/python3.8/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='setup.icloud.com', port=443): Max retries exceeded with url: /setup/ws/1/login?clientBuildNumber=17DHotfix5&clientMasteringNumber=17DHotfix5&ckjsBuildVersion=17DProjectDev77&ckjsVersion=2.0.5&clientId=4C00EFFA-1AD4-11EC-B07B-0242AC110003 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f43e794bb50>: Failed to establish a new connection: [Errno -3] Try again'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/icloud", line 33, in <module>
    sys.exit(load_entry_point('pyicloud-ipd==0.10.1', 'console_scripts', 'icloud')())
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/cmdline.py", line 200, in main
    api = pyicloud_ipd.PyiCloudService(
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 193, in __init__
    self.authenticate()
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 209, in authenticate
    req = self.session.post(
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 590, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 68, in request
    response = super(PyiCloudSession, self).request(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='setup.icloud.com', port=443): Max retries exceeded with url: /setup/ws/1/login?clientBuildNumber=17DHotfix5&clientMasteringNumber=17DHotfix5&ckjsBuildVersion=17DProjectDev77&ckjsVersion=2.0.5&clientId=4C00EFFA-1AD4-11EC-B07B-0242AC110003 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f43e794bb50>: Failed to establish a new connection: [Errno -3] Try again'))
2021-09-21 14:06:10 INFO     Correct owner on config directory, if required
2021-09-21 14:06:10 INFO     Correct group on config directory, if required
2021-09-21 14:06:10 INFO     Generate 2FA cookie using password stored in keyring file.
iCloud Password:
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/authentication.py", line 29, in authenticate
    icloud = pyicloud_ipd.PyiCloudService(
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 141, in __init__
    password = get_password_from_keyring(apple_id)
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/utils.py", line 40, in get_password_from_keyring
    raise NoStoredPasswordAvailable(
pyicloud_ipd.exceptions.NoStoredPasswordAvailable: No pyicloud password for geoffroy.mail@gmail.com could be found in the system keychain.  Use the `--store-in-keyring` command-line option for storing a password for this username.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 353, in connect
    conn = self._new_conn()
  File "/usr/lib/python3.8/site-packages/urllib3/connection.py", line 181, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f09a5d8ee50>: Failed to establish a new connection: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/lib/python3.8/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='setup.icloud.com', port=443): Max retries exceeded with url: /setup/ws/1/login?clientBuildNumber=17DHotfix5&clientMasteringNumber=17DHotfix5&ckjsBuildVersion=17DProjectDev77&ckjsVersion=2.0.5&clientId=5F6FD9FC-1AD4-11EC-B5A5-0242AC110003 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f09a5d8ee50>: Failed to establish a new connection: [Errno -3] Try again'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/icloudpd", line 33, in <module>
    sys.exit(load_entry_point('icloudpd==1.7.2', 'console_scripts', 'icloudpd')())
  File "/usr/lib/python3.8/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/base.py", line 255, in main
    icloud = authenticate(
  File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/authentication.py", line 36, in authenticate
    icloud = pyicloud_ipd.PyiCloudService(
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 193, in __init__
    self.authenticate()
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 209, in authenticate
    req = self.session.post(
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 590, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/lib/python3.8/site-packages/pyicloud_ipd/base.py", line 68, in request
    response = super(PyiCloudSession, self).request(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='setup.icloud.com', port=443): Max retries exceeded with url: /setup/ws/1/login?clientBuildNumber=17DHotfix5&clientMasteringNumber=17DHotfix5&ckjsBuildVersion=17DProjectDev77&ckjsVersion=2.0.5&clientId=5F6FD9FC-1AD4-11EC-B5A5-0242AC110003 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f09a5d8ee50>: Failed to establish a new connection: [Errno -3] Try again'))
2021-09-21 14:06:43 INFO     Web cookie generated. Sync should now be successful.

**Comment by boredazfcuk** [comment]
Hi,

I suspected the changes would highlight the issue:

```
2021-09-21 14:05:26 INFO LAN IP Address: 172.17.0.3
2021-09-21 14:05:26 INFO Default gateway: 172.17.0.1
2021-09-21 14:05:26 INFO DNS server: 10.0.1.1
2021-09-21 14:05:26 ERROR Cannot find icloud.com IP address. Please check your DNS settings
2021-09-21 14:05:31 ERROR No route to icloud.com found. Please check your container's network settings
```

The problem seems to be with your container's DNS server. It's set to 10.0.1.1, which strikes me as odd. It should default to 127.0.0.11 as this IP operates as a forwarder on your host machine. So your containter asks 127.0.0.11 for an IP address, your host machine forwards that to it's resolver, gets the answer back and then passes that answer to the container. As yours is using 10.0.1.1, then it's likely that it's querying some external DNS server directly. 

A variety of issues could be causing this.... The DNS lookups could be blocked by iptables rules on the host machine... the external server might have an access control list which prevents it from responding to your container... It might not have a route to your container network so doesn't know where to reply.

Try omitting the `--network bridge` part of your creation command. This should use the default network which hopefully sets `127.0.0.11` as the default nameserver.

If the default settings of the default network has been changed like instructions [here](https://docs.docker.com/network/bridge/#configure-the-default-bridge-network) then you will need to create a user defined network bridge using the instructions on the same page [here](https://docs.docker.com/network/bridge/#manage-a-user-defined-bridge) and then connect your container to that network instead using something like:

```
docker create network my_network
```
Then create your container like:
```
docker create
--name icloudpd
--network my_network
--restart unless-stopped
-e user=<sensitive info removed>
-e user_id=1026
-e group=users
-e group_id=100
-e apple_id=<sensitive info removed>
-e authentication_type=web
-e folder_structure={:%Y}
-e auto_delete=true
-e synchronisation_interval=86400
-e TZ=Europe/Paris
-v /volume1/docker/iCloud/config:/config
-v /volume1/photo/iCloud:/home/<sensitive info removed>/iCloud 
boredazfcuk/icloudpd
```

**Comment by Geoff-Hz** [comment]
I try on an other Syno and

2021-09-21 18:31:51 INFO     LAN IP Address: 172.17.0.2
2021-09-21 18:31:51 INFO     Default gateway: 172.17.0.1
2021-09-21 18:31:51 INFO     DNS server: 192.168.1.1
2021-09-21 18:31:51 ERROR    Cannot find icloud.com IP address. Please check your DNS settings
2021-09-21 18:31:56 ERROR    No route to icloud.com found. Please check your container's network settings

Same error
The NAS is on an other place so it's not the same config

**Comment by bauzer714** [comment]
@Geoff-Hz  - can you get to icloud.com on the same network? Are you by chance located in China and blocked by the Great Firewall?

**Comment by Geoff-Hz** [comment]
@bauzer714 icloud.com work on both site. I'm from belgium ;-)

**Comment by bauzer714** [comment]
I also have a synology and it's connected using "bridge network".

Is the NAS successfully connected to the internet?
Do you run other Docker containers or is this the only one?
Do the others run under bridge and connect successfully?
If you're on your phone/computer what is the DNS server your computer uses?

**Comment by Geoff-Hz** [comment]
@bauzer714 
for my syno. I have other dockers running find and under bridge (I have pihole but it's OFF for testing)
from phone/computer I have DNS 10.0.1.1

for the second nas. It the first docker I try to install.

**Comment by bauzer714** [comment]
I haven't read through this entire thread, but here is my setup.

Pihole runs as host
![image](https://user-images.githubusercontent.com/13224936/134215510-2e847121-cda4-428a-b27c-c231be33fdfb.png)


All others run as bridge
![image](https://user-images.githubusercontent.com/13224936/134215481-126169c4-e330-4ff7-a6db-edb4de368640.png)


Then in Control Panel Network I have the manually configure DNS server as the NAS IP (I do this so when piHole is upgrading I can change it to 1.1.1.1 and it doesn't die trying to update the lists).

**Comment by bauzer714** [comment]
If you're still having issues - using docker go to the  terminal (I hope you're still on DSM6 - I've read DSM7 doesn't always work with termina). and click "Launch with command"  and type /bin/sh/ - wait a few seconds for it to open and then try the two commands.

//this one will try with your configured DNS server
nslookup icloud.com 

//this one will try with Cloudflare's dns
nslookup icloud.com 1.1.1.1


![image](https://user-images.githubusercontent.com/13224936/134216656-6b96e464-ebca-45bd-8878-4661f37eec9e.png)

**Comment by Geoff-Hz** [comment]
![image](https://user-images.githubusercontent.com/91124229/134218294-5983ba7c-9f6e-4ed5-83ac-706ebfecef22.png)

**Comment by Geoff-Hz** [comment]
thanks I will do that later. It's dinner time :-)

**Comment by boredazfcuk** [comment]
What happens if you force the container to use the Docker Internal DNS server at container creation time with:
```--dns=127.0.0.11```
or force it to use an external DNS with:
```
--dns=1.1.1.1
```

**Comment by bauzer714** [comment]
@boredazfcuk  - with Synology I don't think people are using the docker commands when they create or start containers. Synology offers a pretty good UI that is "good enough".

I'm thinking he has an incorrect value somewhere on his NAS.  He created a shell and executed a nslookup to 1.1.1.1 and it came back unreachable. This was after you guys had additional networks created though. 

Here he received what I think is his 10.x router as the DNS.
![image](https://user-images.githubusercontent.com/13224936/134225388-fefd3ac3-fd09-4eed-9762-fb25f21accdd.png)

But then another syno received his 192.x router as the DNS.
![image](https://user-images.githubusercontent.com/13224936/134225581-a1d83d0d-26e5-4d0c-b8c6-bc77a08bd777.png)

I'm not sure if he as these in two different locations or multiple networks.

For us the easiest way is to stay with bridge network. Override the DNS server to Google or Cloudflare and then restart the container..

![image](https://user-images.githubusercontent.com/13224936/134226045-4f24309d-c127-4732-bf7a-d6584ea40682.png)

![image](https://user-images.githubusercontent.com/13224936/134226115-2beb0435-4c1a-4ee1-a5a1-63080192ca8a.png)

**Comment by boredazfcuk** [comment]
> @boredazfcuk - with Synology I don't think people are using the docker commands when they create or start containers. Synology offers a pretty good UI that is "good enough".

Yeah, I'm not familiar with the Synology DSM interface. I came close to buying one last summer, but just re-jigged some of my parts around to eek a bit more performance out of my current setup instead... I also looked at installing Xpenology which is a freebie version of the Synology DSM, but it looks like a fairly complex installation which needs to be installed on bare metal. I was hoping I could just install it on a VM on my laptop for testing, but that was a no go.

>I'm thinking he has an incorrect value somewhere on his NAS. He created a shell and executed a nslookup to 1.1.1.1 and it came back unreachable. This was after you guys had additional networks created though.

I just think he's using the default network bridge. According to the docs, the default network bridge doesn't have automatic DNS resolution and it also doesn't allow packets to travel outside of the network. See [here](https://docs.docker.com/network/bridge/).

The important bits:
```
Differences between user-defined bridges and the default bridge

    User-defined bridges provide automatic DNS resolution between containers.

    Containers on the default bridge network can only access each other by IP addresses, unless you use the --link option, which is considered legacy. On a user-defined bridge network, containers can resolve each other by name or alias.
```
and also:
```
By default, traffic from containers connected to the default bridge network is not forwarded to the outside world. To enable forwarding, you need to change two settings. These are not Docker commands and they affect the Docker host’s kernel.
```

With this in mind, the issues he's seeing are kind of what I'd expect if the default bridge was being used:
1) Docker is not configuring automatic DNS resolution for the container (or we'd see 127.0.0.11 for the DNS server configured by resolv.conf).
2) Data is not being forwarded outside of the default network (which is why icloud.com is not resolved with the 10.0.1.1 DNS server IP).

Are you using the default network bridge on your device? I'm presuming not?

**Comment by Geoff-Hz** [comment]
Hi Guys,

It is neart to work :-)
I put this:
docker create \
--name icloudpd \
--network host \
--dns=1.1.1.1 \

And I can connect to my icloud.

Now the last issues:
2021-09-22 09:46:09 INFO     Waiting for keyring file to be created...          
2021-09-22 09:48:04 INFO     Keyring file exists, continuing.                   
2021-09-22 09:48:04 INFO     Check download directory mounted correctly         
2021-09-22 09:48:04 ERROR    Failsafe file /home/Geoff/iCloud/.mounted file is not present. Waiting for failsafe file to be created... 

What I have to do?

**Comment by boredazfcuk** [comment]
Connect to the container by running `docker exec -it icloudpd /bin/ash` then create the `.mounted` file with `touch /home/Geoff/iCloud/.mounted`

**Comment by Geoff-Hz** [comment]
Thanks @boredazfcuk 

It's there
/home/Geoff/iCloud # ls -a
.  ..  .mounted

But still not working :(

**Comment by boredazfcuk** [comment]
Does it still say it's wiating for the failsafe file to be created? It should automatically detect you've created the file and then the logs will move past that point.

**Comment by Geoff-Hz** [comment]
it say the same error:
ERROR Failsafe file /home/Geoff/iCloud/.mounted file is not present. Waiting for failsafe file to be created...

I was looking on a closed's post and Bauzer714 say this:
You set the download_path environment variable to /iCloud, but map /home/laurent/iCloud.
Option 1 is delete the variable (the default will be /home/$user/iCloud)
Option 2, Set the third volume mapping for the mount path to /iCloud

But I don't understand well.
Post is https://github.com/boredazfcuk/docker-icloudpd/issues/89

**Comment by boredazfcuk** [comment]
The .mounted file is a failsafe mechanism to make sure that the volume is moutned correctly. Once you create the `/home/Geoff/iCloud/.mounted` file, it should spot it within 5 seconds or so, then start the download process.

If you look in the photo/iCloud directory on your NAS, do you see the `.mounted` file? It will be hidden, so you will need to make sure you can see hidden files.

**Comment by Geoff-Hz** [comment]
Ok I created it on photo/iCloud directory
Restart docker and now it's working.

Thanks a lot. It was very complicate for me to understand everything :-)

**Comment by boredazfcuk** [comment]
Awesome. Glad you got it going in the end.

I'm updating the readme file to include info about networking. I think it will be a good idea to advise people to use custom network bridges, due to the stuff in Docker documentation. Hopefully that will assist other people, as I think there is another person having the same issue.

I'll close this off. Any more problems, just open another issue.

**Comment by Geoff-Hz** [comment]
Hi @boredazfcuk and @bauzer714 

I don't want to re-open the post but I would like to share what I just learn:
The real issues was that on synology's firewall we have to do a rules to open range 172.17.0.0.
I did this and config docker with "-- network bridge" then it can connect to icloud.com.

This was a mistake I didn't think about.
Regards

**Comment by bauzer714** [comment]
I'm wondering if this is due to the VPN connection. I don't have anything permitting 172.x in my firewall.

**Comment by Geoff-Hz** [comment]
I'm not on VPN today.
Just did test:

Without rules
![image](https://user-images.githubusercontent.com/91124229/134365521-e7d6e31c-013b-464d-899a-4177b6ebbb83.png)

and with rules
![image](https://user-images.githubusercontent.com/91124229/134365703-38f96e8a-ef3a-4d59-803b-b1b8f4d8261b.png)

;-)

**Comment by bauzer714** [comment]
I'm intrigued.. now I want to know how mine is even working.
